package/MDAnalysis/visualization/streamlines.py

```diff
@@ -172,7 +176,7 @@ def produce_list_indices_point_in_polygon_this_frame(vertex_coord_list):
             list_indices_point_in_polygon.append(index_list_in_polygon)
         return list_indices_point_in_polygon
 
-    def produce_list_centroids_this_frame(list_indices_in_polygon):
+    def produce_list_centroids_this_frame(list_indices_in_polygon: Sequence[int]) -> Sequence[Optional[float]]:
```

Comment:
You want to be broad when annotating the input so you can accept anything valid from the user, but you want to be specific for the output so the user knows exactly what the function returns. Here you seem to be returning a list of iterators.

Also, `Optional` means a value can be `None`, here it looks like it can be the string `"empty"` instead. You need a `Union[float, str]` rather than `Optional`.